packages/flutter/test/painting/transform_utilities_test.dart

```diff
@@ -39,4 +41,34 @@ void main() {
     test.translate(4.0, 8.0);
     expect(MatrixUtils.getAsTranslation(test), equals(const Offset(6.0, 6.0)));
   });
+
+  test('cylindricalProjectionTransform identity', () {
+    final Matrix4 initialState = MatrixUtils.cylindricalProjectionTransform(
+      radius: 0.0,
+      angle: 0.0,
+      perspective: 0.0,
+    );
+
+    expect(initialState, new Matrix4.identity());
+  });
+
+  test('cylindricalProjectionTransform rotate with no radius', () {
+    final Matrix4 initialState = MatrixUtils.cylindricalProjectionTransform(
+      radius: 0.0,
+      angle: pi / 2.0,
+      perspective: 0.0,
+    );
+
+    expect(initialState, new Matrix4.rotationX(pi / 2.0));
+  });
+
+  test('cylindricalProjectionTransform radius does not change scale', () {
+    final Matrix4 initialState = MatrixUtils.cylindricalProjectionTransform(
+      radius: 1000000.0,
+      angle: 0.0,
+      perspective: 0.0,
+    );
+
+    expect(initialState, new Matrix4.identity());
+  });
```

Comment:
you should probably test this with something that you actually want to make sure works (e.g. a reasonable radius and an angle that isn't a right-angle, with a reasonable perspective).

Reply:
Added a numerical test